message/messaging_service.cc

```diff
@@ -380,19 +380,23 @@ messaging_service::messaging_service(config cfg, scheduling_config scfg, std::sh
         _connection_index_for_tenant.push_back({_scheduling_config.statement_tenants[i].sched_group, i});
     }
 
-    register_handler(this, messaging_verb::CLIENT_ID, [] (rpc::client_info& ci, gms::inet_address broadcast_address, uint32_t src_cpu_id, rpc::optional<uint64_t> max_result_size) {
+    register_handler(this, messaging_verb::CLIENT_ID, [] (rpc::client_info& ci, gms::inet_address broadcast_address, uint32_t src_cpu_id, rpc::optional<uint64_t> max_result_size, rpc::optional<locator::host_id> host_id) {
         ci.attach_auxiliary("baddr", broadcast_address);
         ci.attach_auxiliary("src_cpu_id", src_cpu_id);
         ci.attach_auxiliary("max_result_size", max_result_size.value_or(query::result_memory_limiter::maximum_result_size));
+        ci.attach_auxiliary("host_id", host_id.value_or(locator::host_id::create_null_id()));
```

Comment:
Why do we need host_id on the server side?

The idea in https://github.com/scylladb/scylladb/issues/6403 was to have it on the client side.

Reply:
To know where the message is coming from.

Reply:
That makes sense.

But what I was thinking here was that if we want to address by host_id, we need to know the host id of the server on the client side. This is the part which this series does not address. Basically, the host id in msg_addr is ignored on the sending side, right? In that case, what's the value of adding it?